aioapns/connection.py

```diff
@@ -266,6 +267,17 @@ def on_response_received(self, headers: Dict[bytes, bytes]) -> None:
     def on_data_received(self, raw_data: bytes, stream_id: int) -> None:
         data = json.loads(raw_data.decode())
         reason = data.get("reason", "")
+        timestamp = data.get("timestamp", None)
+        if timestamp:
+            try:
+                # According to the docs "timestamp" is
+                # "represented in milliseconds since Epoch"
+                # so we divide by 1000.0 to get the POSIX
+                # time (seconds) but still keep precision
+                timestamp = datetime.fromtimestamp(int(timestamp) / 1000.0)
```

Comment:
I think it would be safer to use timezone-aware datetime:
```python
timestamp = datetime.fromtimestamp(int(timestamp) / 1000.0, timezone.utc)
```

Reply:
Also, why do we need `int(timestamp)` here? If we get a floating number from APNS then doing so we will lose precision (datetime objects use microseconds), but if we get an integer number then there is no point in converting it to `int`.

Reply:
You're right. I was being extra careful here if by any chance the timestamp was being passed as string but I'm pretty sure we can expect an int here. I'll remove this 😊

Reply:
Do you think we should rather pass this through as int rather than convert it to datetime?

Reply:
> Do you think we should rather pass this through as int rather than convert it to datetime?

I think `datetime` objects are good here. Everyone who needs a timestamp can get it with `timestamp()` method. And since Apple says to include this key only when the error in the :status field is 410, we won't lose much in performance.

Reply:
> Do you think we should rather pass this through as int rather than convert it to datetime?

Actually, this attribute is named as `timestamp`, so it's worth passing through. What do you think? I'm ok to get rid of datetime here.

Reply:
I agree. I’ve updated the code :)